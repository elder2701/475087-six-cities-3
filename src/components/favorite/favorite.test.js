import React from "react";
import renderer from "react-test-renderer";
import Favorite from "./favorite.jsx";

const favorites = {
  favorites: {
    Amsterdam: {
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13
        }
      },
      offers: [
        {
          images: [``, ``],
          title: `Test`,
          rating: 2.1,
          type: `apartment`,
          bedrooms: 5,
          price: 327,
          goods: [`test1`, `test2`],
          description: `test`,
          location: {
            latitude: 52.37854,
            longitude: 4.894976,
            zoom: 16
          },
          id: 5,
          previewImage: ``,
          isFavorite: true,
          isPremium: false,
          maxAdults: 7,
          hostId: 25,
          hostName: `Angelina`,
          hostIsPro: true,
          hostAvatarUrl: ``
        }
      ]
    }
  }
};

it(`<Favorite /> sould be render`, () => {
  const tree = renderer
    .create(<Favorite favorites={favorites} updateStatus={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
