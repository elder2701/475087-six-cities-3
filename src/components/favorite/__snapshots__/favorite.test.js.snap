// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritesList /> sould be render 1`] = `
<article
  className="favorites__card place-card"
>
  <div
    className="favorites__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="110"
        width="150"
      />
    </a>
  </div>
  <div
    className="favorites__card-info place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "NaN%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
      />
    </h2>
    <p
      className="place-card__type"
    />
  </div>
</article>
`;
